### ISSUE TITLE
beforeSuite called in parallel

### ISSUE THREAD

**Issue by phantomato** [Quick/Quick#875]
- [ ] I have read [CONTRIBUTING](https://github.com/Quick/Quick/blob/master/CONTRIBUTING.md) and have done my best to follow them.

### What did you do?

Setup at least two specs each having `beforeSuite`. The `beforeSuite` method sets up the shared state for tests in a spec (e.g. network stubs). 

### What did you expect to happen?

I expected that the `beforeSuite` method should be called in respect to each spec in series. For an example:

1. Run Spec1
2. Setup network stubbing in `beforeSuite`
3. Perform some tests
4. call `afterSuite`
5. Do the same for Spec2

Basically the behaviour of `class func setUp()`

### What actually happened instead?

`beforeSuite` is called for all specs at the same time. If one spec sets up the state for tests, then the next spec can accidentally reset that state by calling it's own `beforeSuite` before the first spec actually started testing.

### Environment

List the software versions you're using:

 - Quick: *2.1.0*
 - Xcode Version: *10.2*
 - Swift Version: *5.0*

**Comment by ikesyo** [comment]
This is a duplicate of #198.

**Comment by phantomato** [comment]
> This is a duplicate of #198. Those run before/after _the entire test suite_ so that is expected behavior.

So that means there is currently no way to setup the shared environment for the spec, besides calling `class func setUp()`?

**Comment by ikesyo** [comment]
CMIIW but you should be able to use `class func setUp()` and `class func tearDown()` since `QuickSpec` extends `XCTestCase`.
